src/factories/store/factoriesSlice.ts

```diff
@@ -11,6 +11,9 @@ export const factoriesSlice = createSlice({
     factories: {},
   } as FactoriesSlice,
   actions: {
+    updateFactories: (fn: (factory: Factory) => void) => state => {
+      Object.values(state.factories).forEach(fn);
```

Comment:
Could we change this to explicit callback? (I don't want to seem pedantic, it's just I fear that one day 😄 
 https://jakearchibald.com/2021/function-callback-risks/ )